File: Transport/@stateflows/common/src/events/behavior-status.notification.ts
import { Notification } from "./notification";
import { BehaviorStatus } from "../enums/behavior-status";

export class BehaviorStatusNotification extends Notification {
    constructor(
        public behaviorStatus: BehaviorStatus,
    ) {
        super();
    }

    public static notificationName = "Stateflows.Common.BehaviorStatusNotification";
}

File: Core/Stateflows.Common/Events/BehaviorStatusNotification.cs
﻿using Newtonsoft.Json;
using System.Collections.Generic;

namespace Stateflows.Common
{
    public class BehaviorStatusNotification : Notification
    {
        public BehaviorStatus BehaviorStatus { get; set; }

        [JsonProperty(TypeNameHandling = TypeNameHandling.None)]
        public IEnumerable<string> ExpectedEvents { get; set; }
    }
}
